molecule/default/roles/k8scopy/tasks/test_copy_large_file.yml

```diff
@@ -0,0 +1,120 @@
+---
+- name: test copy of large binary and text files
+  block:
+  - set_fact:
+      test_directory: "/tmp/test_k8scp_large_files"
+    no_log: true
+
+  - name: create temporary directory for local files
+    ansible.builtin.file:
+      path: "{{ test_directory }}"
+      state: directory
+
+  - name: Download kind executable
+    get_url:
+      url: https://kind.sigs.k8s.io/dl/v0.11.1/kind-linux-amd64
+      dest: "{{ test_directory }}/kind"
+
+  - name: make kind executable
+    ansible.builtin.file:
+      path: "{{ test_directory }}/kind"
+      mode: "+x"
+
+  - name: Run executable from local host
+    command: "{{ test_directory }}/kind version"
+    register: local_result_0
+
+  - name: create large text file
+    k8s_create_file:
+      path: "{{ test_directory }}/large_text_file.txt"
+      size: 1
+
+  # Copy large text file from/to local filesystem to Pod
+  - name: copy large file into remote Pod
+    k8s_cp:
+      namespace: '{{ copy_namespace }}'
+      pod: '{{ pod_with_one_container.name }}'
+      remote_path: /large_text_file.txt
+      local_path: "{{ test_directory }}/large_text_file.txt"
+      state: to_pod
+
+  - name: Compare files
+    kubectl_file_compare:
+      namespace: '{{ copy_namespace }}'
+      pod: '{{ pod_with_one_container.name }}'
+      remote_path: /large_text_file.txt
+      local_path: "{{ test_directory }}/large_text_file.txt"
+      kubectl_path: "{{ kubectl_path }}"
+
+  - name: copy large file from Pod into local filesystem
+    k8s_cp:
+      namespace: '{{ copy_namespace }}'
+      pod: '{{ pod_with_one_container.name }}'
+      remote_path: /large_text_file.txt
+      local_path: "{{ test_directory }}/large_text_file_from_pod.txt"
+      state: from_pod
+
+  - name: Compare files
+    kubectl_file_compare:
+      namespace: '{{ copy_namespace }}'
+      pod: '{{ pod_with_one_container.name }}'
+      remote_path: /large_text_file.txt
+      local_path: "{{ test_directory }}/large_text_file_from_pod.txt"
+      kubectl_path: "{{ kubectl_path }}"
+
+  # Copy large executable file from/to local filesystem to Pod
+  - name: copy large file into remote Pod
```

Comment:
kind is not a particularly large binary file. I'd suggest instead just creating a random file with something like:
```command: dd if=/dev/urandom of={{ local_path }} bs=1M count=100``` or similar.

Rather than testing the output of the executable we should just be checking the sha1sum of the file. It's possible an executable could still function well enough even if some of it got corrupted in the transfer, especially if we're just checking the version.

Reply:
I have created a test module to compare the content, I will keep it for executable to compare both content and execution output